AprilSylph/XKit-Rewritten
# Code correctness fixes from typing experiment

### Description


As per our discussion (https://github.com/AprilSylph/XKit-Rewritten/pull/1234#discussion_r1341964636) in #1234, this fixes a few places where we accidentally passed non-array values to `showModal`, something which would have been caught by a type checker, and adjusts a few JSDoc entries that said type checker would have complained about.

To do this, I—naturally—tried setting up `tsc` on the repository, because come on. Who do you think I am?

### Testing steps

The only functional change expected here, because of the fact that one can use a spread operator on a string, is that Limit Checker's error handler no longer crashes. Thus:
- Open Tumblr with Limit Checker enabled.
- Disconnect your network.
- Click the limit checker button.

With this PR there should be an error message; without it the progress screen should stall forever.
